File: 2024/introducao_programacao/2024.1/07-06-2024/analise.py
s = list(input().split())
letras = input()
dicio = {}
for palavra in s:
    dicio[palavra] = 0
for palavra in s:
    for i in palavra:
        for j in letras:
            if i == j:
                dicio[palavra] += 1
               
ans = 0
for palavra in s:
    if (dicio[palavra] > 0):
        ans += 1   
print(ans)      
    

File: 2024/programacao_avancada/2024.1/07-06-2024/analise.cpp
#include <bits/stdc++.h>

using namespace std;

int main(){
    string s, letras;
    getline(cin, s);
    cin >> letras;
    int ans = 0;
    stringstream palavras(s);
    string palavra;
    bool achou = false;
    map<string, int> mp;
    while (palavras >> palavra){
        for (char i: palavra){
            for (char j: letras){
                if (i == j){
                    mp[palavra]++;
                }
            }
        }
    }
    stringstream palavras2(s);
    while (palavras2 >> palavra){
        if (mp[palavra] > 0){
            ans++;
        }
    }
    cout << ans << "\n";
    return 0;
}